src/main/java/com/google/sps/servlets/HandlePhotoShoppingServlet.java

```diff
@@ -77,36 +75,48 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     // Call GoogleShoppingQuerier to return the extracted products data.
     // First, build the shopping query input.
-    String shoppingQuery;
+    List<String> shoppingQueries = new ArrayList<>();
     try {
-      shoppingQuery = getQuery(request.getParameter("photo-category"), uploadedImageBytes);
+      shoppingQueries = getQuery(request.getParameter("photo-category"), uploadedImageBytes);
     } catch (IllegalArgumentException | PhotoDetectionException exception) {
       response.sendError(SC_INTERNAL_SERVER_ERROR, exception.getMessage());
       return;
     }
 
-    ShoppingQueryInput shoppingQueryInput =
-        new ShoppingQueryInput.Builder(shoppingQuery).language("en").maxResultsNumber(24).build();
+    int maxResultsNumber = 24;
+    if (shoppingQueries.size() > 1) {
```

Comment:
I'd add a comment here for better readability, something like:
```suggestion
    // For shopping-list, where the number of queries exceeds 1, limit the number of products displayed for each item.
    if (shoppingQueries.size() > 1) {
```